### ISSUE TITLE
how to predict image with cuda

### ISSUE THREAD

**Issue by futureflsl** [Deci-AI/super-gradients#1211]
### 🐛 Describe the bug

I use following code can predict image,but it run with cpu
import torch
from super_gradients.common.object_names import Models
from super_gradients.training import models
from PIL import Image
import numpy as np
from super_gradients.training.processing import DetectionCenterPadding, StandardizeImage, ImagePermute, \
    ComposeProcessing, DetectionLongestMaxSizeRescale
checkpoint_path = 'weights/yolo_nas_s_coco.pth'
model = models.get(Models.YOLO_NAS_S, checkpoint_path=checkpoint_path, num_classes=80)
image = Image.open("data/bus.jpg")
names = ['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush']

model.set_dataset_processing_params(
    class_names=names,
    image_processor=ComposeProcessing(
        [
            DetectionLongestMaxSizeRescale(output_shape=(636, 636)),
            DetectionCenterPadding(output_shape=(640, 640), pad_value=114),
            StandardizeImage(max_value=255.0),
            ImagePermute(permutation=(2, 0, 1)),
        ]
    ),
    iou=0.35, conf=0.25,
)
predictions = model.predict(image, conf=0.45, iou=0.45)
predictions.show()

When I try let model to device,it will detect nothing.it seem like image not in cuda.here is the code

import torch
from super_gradients.common.object_names import Models
from super_gradients.training import models
from PIL import Image
import numpy as np
from super_gradients.training.processing import DetectionCenterPadding, StandardizeImage, ImagePermute, \
    ComposeProcessing, DetectionLongestMaxSizeRescale

checkpoint_path = 'weights/yolo_nas_s_coco.pth'
model = models.get(Models.YOLO_NAS_S, checkpoint_path=checkpoint_path, num_classes=80)
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model.to(device)
image = Image.open("data/bus.jpg")

names = ['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush']

model.set_dataset_processing_params(
    class_names=names,
    image_processor=ComposeProcessing(
        [
            DetectionLongestMaxSizeRescale(output_shape=(636, 636)),
            DetectionCenterPadding(output_shape=(640, 640), pad_value=114),
            StandardizeImage(max_value=255.0),
            ImagePermute(permutation=(2, 0, 1)),
        ]
    ),
    iou=0.35, conf=0.25,
)
predictions = model.predict(image, conf=0.45, iou=0.45)
predictions.show()

so how to modify this code,let code run with cuda?

### Versions

windows10
python3.8
super-gradients==3.1.2

**Comment by Louis-Dupont** [comment]
Hi @futureflsl ,
Did you make sure that your device supports cuda? You can check if `device == cpu` or == `device == cuda`.
If `device==cuda` then you can check if your model was properly moved to cuda.
Our predict function works using the model device, so if your model is already on `cuda` there is no apparent reason for it to run on `cpu`.

Besides, I saw you are using a more complex approach to the predict method which, as explained [here](https://docs.deci.ai/super-gradients/documentation/source/PredictionSetup.html) is meant only for if you trained a model using a dataset that doesnt inherit from any of the SG dataset.
**As long as you are working with pretrained weights, or that you are using a model trained using any of our dataset, you don't need to do all these setup steps and can stick to the following code** (as described [here](https://docs.deci.ai/super-gradients/documentation/source/ModelPredictions.html)):
``` python
from super_gradients.common.object_names import Models
from super_gradients.training import models

model = models.get(Models.YOLO_NAS_S, checkpoint_path="weights/yolo_nas_s_coco.pth", num_classes=80).cuda()

predictions = model.predict("data/bus.jpg", conf=0.45, iou=0.45)
predictions.show()
```

Besides, if this weights `"weights/yolo_nas_s_coco.pth"` is actually the pretrained weights of our model zoo, then you can remove `checkpoint_path` and `num_classes` and just set `pretrained_weights="coco"`:
``` python
from super_gradients.common.object_names import Models
from super_gradients.training import models

model = models.get(Models.YOLO_NAS_S, pretrained_weights='coco').cuda()

predictions = model.predict("data/bus.jpg", conf=0.45, iou=0.45)
predictions.show()
```

Hoping this helps

**Comment by Louis-Dupont** [comment]
We have a [tutorial](https://docs.deci.ai/super-gradients/documentation/source/ModelPredictions.html) that covers the main questions going around the `model.predict(...)`, it includes among others getting access to each individual bbox and predicted class.
Let me know if anything is unclear

**Comment by futureflsl** [comment]
super-gradient==3.1.2 cuda detect nothing,I downgrade version to 3.1.0,it wok well

**Comment by SheDevil123** [comment]
> super-gradient==3.1.2 cuda detect nothing,I downgrade version to 3.1.0,it wok well

how were u able to downgrade to 3.1.0???
i keep getting an error saying its unable to build wheel for pycocotools 2.0.4
super_gradients 3.1.2 and above uses pycocotools 2.0.6 which is installs fine
pls tell me how u downgraded
